### PR TITLE
Mission and tutorial

### PR DESCRIPTION
**Pull request by tomsoftware**
* add tutorial function
* fix bug in [game.cc -> bool game_t::load_mission_map(int level) ] causes wrong mission-random is used
* fix error in is_valid_extension() if the road has zero length (may only occurs in MS Studio)

### CODE REVIEW

**Comment by wdigger** [comment]
Please, merge commits by themes - data loading, game init and so on ( https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/ ).

**Comment by tomsoftware** [comment]
ok... here is the mission and tutorial stuff and some fixes I need to run this

**Comment by wdigger** [comment]
1. Hint how to rebase properly: https://robots.thoughtbot.com/keeping-a-github-fork-updated
2. I'm not sure is tutorial games are needed at game init window at this time, we have no support of such game mode in game logic, not all buildings preparing when tutorial starts and many more problems.

If you really interested in tutorial implementation, try to play tutorials in original game, read game user manual, and realize appropriate roles in game code.

3. Now it's too complicated to separate fixes to stand-alone commit, I may take it manually, or you may try to do it yourself. For training you git skills :)

**Comment by wdigger** [comment]
If you need a user manual, please email me to wicked_digger@mail.ru

**Comment by tomsoftware** [comment]
May I'm wrong but the only difference between a normal Game and Training is that Training is finished if some conditions are reached. E.g.
1. Build one Hut, Tower and Castle
2. Build one Forest house, lumberjack, saw mill, and a stonecutter
3. Produce food
4. Produce gold
5. Produce tools and weapons
6. Fight
But there are no special in game/GUI feature required.

The main aim of this commit is to show a way how to manage more the random game and mission... the current way how this is done is by looking at the "game_mission" variable. >0 mission; <0 random ... this is not a good way how to handle this.

**Comment by tomsoftware** [comment]
OK... is see your point: this feature needs some more work!
=> so I did separate the fixes to another pull request
=> I add victory_condition to the "src/mission.h"
=> src/game.cc now calls the new player_check_victory_conditions() function to check if the play satisfy those
Is this a good way of doing this?

File: src/player.h [hunk 1]
```diff
@@ -225,7 +226,7 @@ class player_t : public game_object_t {
   void decrease_serf_count(int type) { serf_count[type]--; }
   int *get_serfs() { return reinterpret_cast<int*>(serf_count); }
```
Comment (tomsoftware) [inline src/player.h]: du you know what the serf_count[] is and why increase_res_count() is increasing this as well as decrease_serf_count() did?

Comment (wdigger) [inline src/player.h]: It's a bug. Result of copy-past.

**Pull request closed without merging**